builder/common/step_key_pair.go

```diff
@@ -12,12 +12,16 @@ import (
 	"github.com/hashicorp/packer-plugin-sdk/multistep"
 	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
 	"github.com/hashicorp/packer-plugin-sdk/retry"
+	"github.com/hashicorp/packer-plugin-sdk/template/interpolate"
 )
 
 type StepKeyPair struct {
 	Debug        bool
 	Comm         *communicator.Config
 	DebugKeyPath string
+	IsRestricted bool
```

Comment:
While I agree this could be changed, the name is used throughout Packer.
I suggest we keep it as is in this PR and make a new one later, changing all the occurences to something like `IsRegionRestricted` instead.
What do you think?

Reply:
Sounds like a good way forward to me. Thanks for thinking a little deeper on this suggestion.